src/TextField/TextField.js

```diff
@@ -47,6 +47,7 @@ const getStyles = (props, context, state) => {
     },
     floatingLabel: {
       color: hintColor,
+      pointerEvents: 'none',
```

Comment:
Well, the solution that we will came up with needs to address at least the three issues we have raised.
Here his what came into my mind:
 - Use a `onTouchTap` on the `floatingLabel` to address https://github.com/callemall/material-ui/pull/4253, https://github.com/callemall/material-ui/issues/3757.
 - Reduce the *height* of the `input` in the `TextField` component to match the text height.
 - For https://github.com/callemall/material-ui/issues/3908, we could move `onTouchTap` to the root component of the `TextField`.
 - For https://github.com/callemall/material-ui/issues/4218, We could make the `DropDownMenu` controllable and let the `SelectField` control it.

**TL;DR** `pointerEvents` sounds like a good quick win.